socorro/unittest/processor/rules/test_mozilla.py

```diff
@@ -1384,90 +1384,100 @@ def test_aurora_channel(self):
-        assert processed_crash["os_pretty_version"] == "OS X Unknown"
+        assert processed_crash["os_pretty_version"] == "Ubuntu 18.04 LTS"
 
-        # Any version of Linux.
-        processed_crash = DotDict()
-        processed_crash.os_name = "Linux"
-        processed_crash.os_version = "0.0.12.13"
+    @pytest.mark.parametrize(
+        "os_name, os_version, expected",
+        [
+            ("Linux", None, "Linux"),
+            (None, None, None),
+            ("Windows NT", "NaN", "Windows NT"),
+        ],
+    )
+    def test_junk_data(self, os_name, os_version, expected):
+        raw_crash = copy.deepcopy(canonical_standard_raw_crash)
+        raw_dumps = {}
+        processor_meta = get_basic_processor_meta()
 
-        rule.act(raw_crash, raw_dumps, processed_crash, processor_meta)
-        assert "os_pretty_version" in processed_crash
-        assert processed_crash["os_pretty_version"] == "Linux"
+        rule = OSPrettyVersionRule()
 
         # Now try some bogus processed_crashes.
-        processed_crash = DotDict()
-        processed_crash.os_name = "Lunix"
-        processed_crash.os_version = None
+        processed_crash = {}
+        if os_name is not None:
+            processed_crash["os_name"] = os_name
+        if os_version is not None:
+            processed_crash["os_version"] = os_version
 
         rule.act(raw_crash, raw_dumps, processed_crash, processor_meta)
-        assert "os_pretty_version" in processed_crash
-        assert processed_crash["os_pretty_version"] == "Lunix"
+        assert processed_crash["os_pretty_version"] == expected
 
-        processed_crash = DotDict()
-        processed_crash.os_name = None
-        processed_crash.os_version = None
-
-        rule.act(raw_crash, raw_dumps, processed_crash, processor_meta)
-        assert "os_pretty_version" in processed_crash
-        assert processed_crash["os_pretty_version"] is None
+    def test_dotdict(self):
+        raw_crash = copy.deepcopy(canonical_standard_raw_crash)
+        raw_dumps = {}
+        processor_meta = get_basic_processor_meta()
 
-        processed_crash = DotDict()
+        rule = OSPrettyVersionRule()
 
+        processed_crash = DotDict(
+            {"os_name": "Windows NT", "os_version": "10.0.11.7600"}
+        )
         rule.act(raw_crash, raw_dumps, processed_crash, processor_meta)
-        assert "os_pretty_version" in processed_crash
-        assert processed_crash["os_pretty_version"] is None
+        assert processed_crash["os_pretty_version"] == "Windows 10"
 
-        processed_crash = DotDict()
-        processed_crash.os_name = "Windows NT"
-        processed_crash.os_version = "NaN"
+    def test_none(self):
+        raw_crash = copy.deepcopy(canonical_standard_raw_crash)
+        raw_dumps = {}
+        processor_meta = get_basic_processor_meta()
 
+        rule = OSPrettyVersionRule()
+
+        processed_crash = {"os_name": None, "os_version": None}
```

Comment:
In the earlier tests, if `os_name` or `os_version` were `None`, then they were omitted from the `processed_crash`. So this explicitly tests what happens if the values were `None` rather than omitted.